nasl/nasl_frame_forgery.c

```diff
@@ -0,0 +1,183 @@
+/* Copyright (C) 2021 Greenbone Networks GmbH
+ *
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+/**
+ * @file nasl_frame_forgery.c
+ * @brief Functions to forge and manipulate datalink layer frames.
+ */
+
+#include "nasl_frame_forgery.h"
+
+#include "../misc/bpf_share.h" /* for bpf_open_live */
+#include "../misc/plugutils.h"
+#include "capture_packet.h"
+
+#include <gvm/base/networking.h>
+#include <libnet.h>
+#include <net/ethernet.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#undef G_LOG_DOMAIN
+/**
+ * @brief GLib logging domain.
+ */
+#define G_LOG_DOMAIN "lib  misc"
+
+
+/**
+ * @brief Send an arp request to an IP host.
+ *
+ * @naslnparam
+ *
+ * - @a host    Target's IPv4 address
+ *
+ * @naslret The MAC address of the host. NULL otherwise
+ *
+ * @param[in] lexic   Lexical context of NASL interpreter.
+ *
+ * @return A tree cell.
+ */
+tree_cell *
+nasl_send_arp_request (lex_ctxt *lexic)
+{
+  tree_cell *retc;
+  struct in6_addr *dst = plug_get_host_ip (lexic->script_infos);
+  struct in_addr inaddr;
+  char ip_str[INET6_ADDRSTRLEN];
+  libnet_t *l; /* the libnet context */
+  char errbuf[LIBNET_ERRBUF_SIZE];
+  u_int32_t target_ip_addr, src_ip_addr;
+  u_int8_t mac_broadcast_addr[6] = {0xff, 0xff, 0xff, 0xff, 0xff, 0xff};
+  u_int8_t mac_zero_addr[6] = {0x0, 0x0, 0x0, 0x0, 0x0, 0x0};
+  struct libnet_ether_addr *src_mac_addr;
+  int bytes_written;
+
+  l = libnet_init (LIBNET_LINK, NULL, errbuf);
+  if (l == NULL)
+    {
+      fprintf (stderr, "libnet_init() failed: %s\n", errbuf);
+      exit (EXIT_FAILURE);
```

Comment:
Why use `fprintf` instead of the usual logging facility?
Why `exit`? Is returning NULL not enough?
Same for all other cases in this file.